### PR TITLE
[E-shop][Layout][Task] Create basic layout for all section

### PR DESCRIPTION
**Pull request by st-ngabui**
**Description:**
- Create file html and style for homepage use SASS.

**Dependence: **
#8

### CODE REVIEW

File: excercise-e-shop/home.html [hunk 1]
```diff
@@ -0,0 +1,356 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
```
Comment (st-hahoang) [inline excercise-e-shop/home.html:7]: Sửa lại title

File: excercise-e-shop/home.html [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  <header>
+    <div class="container">
+      <div class="header-left">
+        <h1>
+          <a href="#" class="logo">
+            <img src="#" alt="logo">
+          </a>
+        </h1>
+      </div>
+      <div class="header-center">
```
Comment (st-hahoang) [inline excercise-e-shop/home.html:19]: Xem xét dùng thẻ của html5 cho phần menu.

File: excercise-e-shop/home.html [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        <button class="search">
+          <i class="icon"></i>
+        </button>
+        <button class="cart">
+          <i class="icon"></i>
+        </button>
+        <button class="user">
+          <i class="icon"></i>
+        </button>
+      </div>
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:34]: consider `ul > li` and contain `a` tag

File: excercise-e-shop/home.html [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+    </div>
+  </header>
+  <main>
+    <section class="banner">
+      <p class="banner-title">Sale of the <span>summer</span>collection</p>
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:39]: `heading` tag

File: excercise-e-shop/home.html [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+            </div>
+          </li>
+        </ul>
+      </div>
+    </section>
+    <section>
+      <div class="container">
+        <div class="recommend-cloth-title">
+          <h4 class="text-title">Selected just for you</h4>
+          <a href="#" class="btn btn-secondary">SHOW MORE</a>
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:105]: `SHOW MORE` should be `position: absolute` from this `section` due to this action show more information for all section, not section-header

File: excercise-e-shop/home.html [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+        </div>
+        <ul class="grid-list">
+          <li class="grid-item product">
+            <div class="product-item">
+              <a href="#" class="product-info badge-sale">
+                <img src="#" alt="product">
+                <h4 class="product-name">T-Shirt Summer Vibes</h4>
+              </a>
+              <p class="product-price badge-price-sale">$89.99</p>
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:114]: where is price before sale

File: excercise-e-shop/home.html [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+              </a>
+              <p class="product-price badge-price-sale">$119.99</p>
+            </div>
+          </li>
+        </ul>
+      </div>
+    </section>
+    <section>
+      <div class="container">
+        <h4 class="text-title text-center">
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:149]: `section-title` will make sense than `text-title`

File: excercise-e-shop/home.html [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+                <img src="#" alt="product">
+                <h4 class="product-name">Loose Textured T-Shirt</h4>
+              </a>
+              <p href="#" class="product-price badge-price-sale">$119.99</p>
+            </div>
+          </li>
+        </ul>
+      </div>
+    </section>
+    <section>
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:253]: need a class name for this section to define `img/background`

File: excercise-e-shop/home.html [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+            <a href="#" class="social-icon">
+              <i class="icon"></i>
+            </a>
+            <a href="#" class="social-icon">
+              <i class="icon"></i>
+            </a>
+            <a href="#" class="social-icon">
+              <i class="icon"></i>
+            </a>
+          </div>
```
Comment (st-quocpham) [inline excercise-e-shop/home.html:290]: `ul > li`

File: excercise-e-shop/assets/scss/base/_all.scss [hunk 10]
```diff
@@ -0,0 +1,4 @@
+@import './reset';
+@import './font';
+@import './variable';
+@import './mixin';
```
Comment (st-quocpham) [inline excercise-e-shop/assets/scss/base/_all.scss:4]: all files need a blank line at the end.

File: excercise-e-shop/assets/scss/base/_font.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+body {
+  font-family: 'poppins', sans-serif;
+}
```
Comment (st-quocpham) [inline excercise-e-shop/assets/scss/base/_font.scss:3]: need more detail: font-size, line-height, ...

File: excercise-e-shop/assets/scss/base/_mixin.scss [hunk 12]
```diff
@@ -0,0 +1,11 @@
+@mixin badge-sale($left) {
```
Comment (st-quocpham) [inline excercise-e-shop/assets/scss/base/_mixin.scss:1]: `mixin` need default value for arguments.
example: `@mixin badge-sale($left: left)`

File: excercise-e-shop/assets/scss/base/_variable.scss [hunk 13]
```diff
@@ -0,0 +1,7 @@
+$base-color: #FBB03B;
```
Comment (st-quocpham) [inline excercise-e-shop/assets/scss/base/_variable.scss:1]: not only color, font-size, font-family, break-points, every reusable-value need to be defined by variables

**Review by st-quocpham — COMMENTED** [review]
- Break down css part of `Home` too detail sections in order to easier maintenance.
- Need to define base part more detail
- Apply `function/feature` of scss/sass

**Pull request merged**